node/pkg/governor/governor.go

```diff
@@ -179,12 +183,15 @@ func (ce *chainEntry) isBigTransfer(value uint64) bool {
 }
 
 type ChainGovernor struct {
-	db                    db.GovernorDB // protected by `mutex`
-	logger                *zap.Logger
-	mutex                 sync.Mutex
-	tokens                map[tokenKey]*tokenEntry     // protected by `mutex`
-	tokensByCoinGeckoId   map[string][]*tokenEntry     // protected by `mutex`
-	chains                map[vaa.ChainID]*chainEntry  // protected by `mutex`
+	db                  db.GovernorDB // protected by `mutex`
+	logger              *zap.Logger
+	mutex               sync.Mutex
+	tokens              map[tokenKey]*tokenEntry    // protected by `mutex`
+	tokensByCoinGeckoId map[string][]*tokenEntry    // protected by `mutex`
+	chains              map[vaa.ChainID]*chainEntry // protected by `mutex`
+	// We maintain a sorted slice of governed chainIds so we can iterate over maps in a deterministic way
+	// This slice should be sorted in ascending order by (Wormhole) Chain ID.
+	chainIds              []vaa.ChainID
```

Comment:
If you make this a slice of `*chainEntry` you would not have to do map look ups everywhere you iterate through it. Of course they you should probably rename it to something like `chainsById`.

Reply:
Do you mean changing `chains` from a map to a slice and removing `chainIds`?

Reply:
No, keep `chains` as is so we can do a map look up by chainID. Just replace `chainIds` with `chainsById` where the payload would be `*chainEntry` so you could use it to directly access the chain entry without doing that extra map lookup.

Reply:
Could we break this out into a separate issue and not have it block the PR?
I see how it might be more optimal to iterate over pointers directly but I think having both data structures also means that we need to keep the pointers in sync.

I tried making this change and it broke a lot of things in the test suite. I think the reason is that there are helper functions there to update the `chainEntry` pointer in the map, which is just fine. But it means every time we do that we also need to update the `chainsById` slice. In the actual governor code we don't update the `chainEntry` pointer in the map but it feels like an accident waiting to happen. 

My preference would be to keep it simple and iterate over a static list of ~30 ints rather than worry about keeping these two things in sync.

Reply:
I do feel that it would be better to eliminate a bunch of unnecessary map look ups, especially in any code called from the main processor routine, but I'm okay with deferring that to a future PR.

Reply:
Thanks I appreciate the flexibility. I'll write up a GitHub Issue summarizing the above